framework/doc/content/source/meshgenerators/TransfiniteMeshGenerator.md

```diff
@@ -0,0 +1,42 @@
+# TransfiniteMeshGenerator
+
+The `TransfiniteMeshGenerator` produces two-dimensional meshes from a set of 4 corner vertices by interpolating the edges into the inner domain. The resulting mesh  consists of QUAD4 elements of aspect ratio and skewness determined by the points on the boundary edges.
+
+!syntax description /Mesh/TransfiniteMeshGenerator
+
+## Overview
+
+!media media/meshgenerators/quad_convention.png style=float:right;width:20%;margin-left:2%; caption=Fig. 1: Convention for the corner vertices order.
+
+Given a set of 4 corners the TransfiniteMeshGenerator constructs by default the straight edges that connect the vertices and fills up the interior with straight lines. The test, [quadrilater_generator.i](test/tests/meshgenerators/transfinite_generator/quadrilater_generator.i),
+illustrates the use of the `TransfiniteMeshGenerator` object to construct general quadrilaterals. The convention for the corners numbering should follow Fig. 1, where we use the alternative sides numbering instead of the top-bottom-left-right as in the input file, and should be prescribed in the order +(0,0) - (0,1) - (1,1) - (1,0)+.
```

Comment:
Thanks for adding this documentation--I've pulled your branch because I'm excited to try out this new capability (which is essential for the stochastic tools module study I'd like to run). Is the convention for the corners number, mentioned in this line, the order in which the points should be given in the input file or the order in which the code builds the quadrilateral?

As more of a user, I'm particularly interested in figuring out what order I should give the points in the input file. For what it's worth, I've followed the order `(0,0) - (0,1) - (1,1) - (1,0)` and that seems to build nicely

Reply:
Yes, the order is what is meant, better to approach it as a convention for now since I am still debating a few things with Derek et co. and I might refine the explanations to be more clear. 

And happy you find it useful, I wasn't sure it would be. I pushed now fixes for all arc circle cases and a test for prescribing discrete points so is still growing.

Reply:
As promised, the MWE/MNWE with the winding order aux kernel. As you'll see, it is all user error (between my keyboard and my chair) that caused the issue, and I'm not sure if it is possible to guard against this sort of user error in the code:
```
[Mesh]
  [quadrilateral]
    type = TransfiniteMeshGenerator
    #Gives a positive winding order which is not good for extruding to 3D
    corners = '-0.6 -0.5 0.0
                0.8 -0.5 0.0
                0.7  0.5 0
               -0.6  0.5 0'

    #Gives a negative winding order and works well for extruding to 3D
    # corners = '-0.6  0.5 0
    #             0.7  0.5 0
    #             0.8 -0.5 0.0
    #            -0.6 -0.5 0.0'
    nx = 4
    ny = 9
    bottom_type = LINE
    top_type = LINE
    right_type = LINE
    left_type = LINE
  []
[]

[AuxVariables]
  [winding_order]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[AuxKernels]
  [winding_order]
    type = WindingOrder
    variable = winding_order
  []
[]

[Variables]
  [u]
  []
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = u
  []
[]

[BCs]
  [left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
  [right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  []
[]

[Postprocessors]
  [min_winding_order]
    type = ElementExtremeValue
    value_type = min
    variable = winding_order
  []
  [max_winding_order]
    type = ElementExtremeValue
    value_type = max
    variable = winding_order
  []
[]

[Executioner]
  type = Steady
[]

[Outputs]
  exodus = true
  csv = true
  hide = winding_order
[]
```

Reply:
@loganharbour would you mark this conversation as resolved for me please? I do not appear to have the privileges to do so